src/config_utils.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Configuration diffing utils."""
+
+import hashlib
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Union
+
+from ops.model import Container
+
+
+def _sha256(hashable) -> int:
```

Comment:
As far as I understand `hashable` is either `str` or `bytes`.

Shouldn't we have this signature?

```suggestion
def _sha256(hashable: Union[str, bytes]) -> int:
```

Reply:
Fixed in next commit.